emgapianns/views.py

```diff
@@ -476,103 +689,158 @@ class AnalysisOrganismRelationshipViewSet(  # NOQA
+    ---
+    `/analyses/MGYA00102827/taxonomy/itsonedb`
+    ---
+    """
+    serializer_class = m_serializers.OrganismRetriveSerializer
 
-        return super(AnalysisOrganismSSURelationshipViewSet, self) \
-            .list(request, *args, **kwargs)
+    pagination_class = m_page.MaxSetPagination
 
+    filter_backends = (
+        filters.OrderingFilter,
+    )
 
-class AnalysisOrganismLSURelationshipViewSet(  # NOQA
-    m_viewsets.ListReadOnlyModelViewSet):
+    ordering_fields = (
+        'name',
+        'prefix',
+        'lineage',
+    )
 
+    lookup_field = 'accession'
+
+    annotation_model = m_models.AnalysisJobTaxonomy
+
+    annotation_model_property = 'taxonomy_itsonedb'
+
+
+class AnalysisOrganismITSUniteRelationshipViewSet(  # NOQA
+    m_mixins.AnalysisJobAnnotationMixin,
+    m_viewsets.ListReadOnlyModelViewSet):
+    """Retrieves ITS UNITE Taxonomic analysis for the given accession
+    Example:
+    ---
+    `/analyses/MGYA00102827/taxonomy/unite`
+    ---
+    """
     serializer_class = m_serializers.OrganismRetriveSerializer
 
     pagination_class = m_page.MaxSetPagination
 
+    filter_backends = (
+        filters.OrderingFilter,
+    )
+
+    ordering_fields = (
+        'name',
+        'prefix',
+        'lineage',
+    )
+
     lookup_field = 'accession'
 
-    def get_queryset(self):
+    annotation_model = m_models.AnalysisJobTaxonomy
+
+    annotation_model_property = 'taxonomy_itsunite'
+
+
+class AnalysisTaxonomyOverview(APIView):
+    """Get the counts for each taxonomic results for an analysis job.
+    """
+
+    def get(self, request, accession):
+        """Get the AnalysisJob and then the AnalysisJobTaxonomy
+        """
         job = get_object_or_404(
             emg_models.AnalysisJob,
-            Q(pk=int(self.kwargs['accession'].lstrip('MGYA')))
+            Q(pk=int(accession.lstrip('MGYA')))
         )
-
         analysis = None
         try:
             analysis = m_models.AnalysisJobTaxonomy.objects \
                 .get(analysis_id=str(job.job_id))
         except m_models.AnalysisJobTaxonomy.DoesNotExist:
-            pass
-
-        return getattr(analysis, 'taxonomy_lsu', [])
-
-    def list(self, request, *args, **kwargs):
-        """
-        Retrieves LSU Taxonomic analysis for the given accession
-        Example:
-        ---
-        `/analyses/MGYA00102827/taxonomy/lsu`
-        """
+            raise Http404
 
-        return super(AnalysisOrganismLSURelationshipViewSet, self) \
-            .list(request, *args, **kwargs)
+        return Response({
+            'accession': analysis.accession,
+            'pipeline_version': analysis.pipeline_version,
+            'taxonomy_count': len(getattr(analysis, 'taxonomy', [])),
+            'taxonomy_ssu_count': len(getattr(analysis, 'taxonomy_ssu', [])),
+            'taxonomy_lsu_count': len(getattr(analysis, 'taxonomy_lsu', [])),
+            'taxonomy_itsunite_count': len(getattr(analysis, 'taxonomy_itsunite', [])),
```

Comment:
Is this a reference to the directory name on the file system? It not we can leave it like that.

Reply:
No, that is on the model.